src/cowboy_websocket.erl

```diff
@@ -411,3 +412,10 @@ handler_terminate(#state{env=Env, handler=Handler},
 		Req, HandlerState, Reason) ->
 	cowboy_handler:terminate(Reason, Req, HandlerState, Handler),
 	{ok, Req, [{result, closed}|Env]}.
+	
+	
+%% @private
+get_real_socket({proxy_socket, _, Socket, _, _, _, _, _, _}) -> Socket;
```

Comment:
It's a new function, messages/0 becomes messages/1 (to have the socket in arguments), and you can still keep a messages/0 around in your deployment for a while to ease transition.

Will link to this thread from the Ranch 2 meta ticket.